README.md

```diff
@@ -56,13 +56,43 @@ parameter set to the directory you want to output the Go code to.
 The generated files will be suffixed .pb.go.  See the Test code below
 for an example using such a file.
 
+## Packages and input paths ##
+
+The protocol buffer language has a concept of "packages" which does not
+correspond well to the Go notion of packages. In generated Go code,
+each source `.proto` file is associated with a single Go package. The
+name and import path for this package is specified with the `go_package`
+proto option:
+
+	option go_package = "github.com/golang/protobuf/ptypes/any";
+
+The protocol buffer compiler will attempt to derive a package name and
+import path if when a `go_package` option is not present, but it is
+best to always specify one explicitly.
+
+The filename of generated files may be produced in one of two ways.
```

Comment:
Perhaps:
> The output name of the generated file is always produced the same way where the `.proto` suffix is replaced with `.pb.go` (e.g., "foo.proto" is named "foo.pb.go"). 
However, the output directory is selected in one of two ways.

Reply:
Phrased a bit differently; PTAL.